tests/test_odp_segment_manager.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2022, Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from unittest import mock
+
+from requests import exceptions as request_exception
+
+from optimizely.helpers.enums import OptimizelySegmentOption
+from optimizely.odp.lru_cache import LRUCache
+from optimizely.odp.odp_config import OdpConfig
+from optimizely.odp.odp_segment_manager import OdpSegmentManager
+from optimizely.odp.zaius_graphql_api_manager import ZaiusGraphQLApiManager
+from tests import base
+
+
+class OdpSegmentManagerTest(base.BaseTest):
+    api_host = 'host'
+    api_key = 'valid'
+    user_key = 'fs_user_id'
+    user_value = 'test-user-value'
+
+    def test_empty_array_with_no_segments_to_check(self):
+        with mock.patch('requests.post') as mock_request_post, \
+                mock.patch('optimizely.logger') as mock_logger:
+            mock_request_post.return_value = self.fake_server_response(status_code=200,
+                                                                       content=self.good_response_data)
+
+            odp_config = OdpConfig(self.api_key, self.api_host, [])
+            segments_cache = LRUCache(1000, 1000)
+            api = ZaiusGraphQLApiManager()
+
+            segment_manager = OdpSegmentManager(odp_config, segments_cache, api, mock_logger)
+            segments = segment_manager.fetch_qualified_segments(self.user_key, self.user_value, [])
+
+            self.assertEqual(segments, [])
+            mock_logger.debug.assert_called_once_with('No segments are used in the project. Returning empty list.')
+            mock_logger.error.assert_not_called()
```

Comment:
Can we also validate that graphQLApi.fetchSegment() is not called in this case?

Reply:
Check added - see line 51 below